tests/AStarTests.elm

```diff
@@ -0,0 +1,96 @@
+module AStarTests exposing (tests)
+
+import AStar exposing (..)
+import Expect exposing (Expectation)
+import Fuzz exposing (..)
+import Set exposing (Set)
+import Test exposing (..)
+
+
+tests : Test
+tests =
+    describe "AStar"
+        [ findPathTests
+        , findPathFuzzTests
+        , straightLineCostFuzzTests
+        ]
+
+
+findPathTests : Test
+findPathTests =
+    describe "findPath"
+        [ test "it finds a path" <|
+            \() ->
+                findPath straightLineCost movesFrom ( 0, 0 ) ( 2, 0 )
+                    |> Expect.equal (Just [ ( 1, 0 ), ( 2, 0 ) ])
+        ]
+
+
+findPathFuzzTests : Test
+findPathFuzzTests =
+    describe "findPathFuzzTests"
+        [ fuzz listOfPositionsWithLowCost "if all moves are available, a path is always found" <|
+            \positions ->
+                positions
+                    |> List.map
+                        (\( start, end ) ->
+                            findPath straightLineCost movesFrom start end
+                        )
+                    |> List.all (\x -> x /= Nothing)
+                    |> Expect.equal True
+        ]
+
+
+straightLineCostFuzzTests : Test
+straightLineCostFuzzTests =
+    describe "straightLineCostFuzzTests"
+        [ fuzz tupleOfPositions "cost is always non-negative" <|
+            \( start, end ) ->
+                straightLineCost start end |> Expect.atLeast 0.0
+        ]
+
+
+
+----------
+
+
+listOfPositionsWithLowCost : Fuzzer (List ( Position, Position ))
+listOfPositionsWithLowCost =
+    listOfPositionsWithCost
+        |> map (List.filter (\( _, cost ) -> cost <= 100.0))
+        |> map (List.map Tuple.first)
```

Comment:
I thought there would be a `Fuzz.filter` but there is not - I'm not sure of a better way to do this. This works for now :)

Reply:
I _think_ you want `floatRange` there...

Reply:
But on the other hand, for this case I don't think it'll matter, and `List.filter` is easier to read.